static/app/views/alerts/details/alertChart.tsx

```diff
@@ -1,28 +1,84 @@
 import * as React from 'react';
 import styled from '@emotion/styled';
 
+import AsyncComponent from 'sentry/components/asyncComponent';
 import AreaChart from 'sentry/components/charts/areaChart';
 import {HeaderTitleLegend, SectionHeading} from 'sentry/components/charts/styles';
+import type {DateTimeObject} from 'sentry/components/charts/utils';
 import {Panel, PanelBody, PanelFooter} from 'sentry/components/panels';
 import Placeholder from 'sentry/components/placeholder';
 import {t} from 'sentry/locale';
 import space from 'sentry/styles/space';
-import {Organization} from 'sentry/types';
+import {Organization, Project} from 'sentry/types';
+import {IssueAlertRule, ProjectAlertRuleStats} from 'sentry/types/alerts';
 import getDynamicText from 'sentry/utils/getDynamicText';
 
-type Props = {
-  orgId: string;
-  organization: Organization;
+type Props = AsyncComponent['props'] &
+  DateTimeObject & {
+    orgId: string;
+    organization: Organization;
+    project: Project;
+    rule: IssueAlertRule;
+  };
+
+type State = AsyncComponent['state'] & {
+  alerts: ProjectAlertRuleStats[];
 };
 
-class AlertChart extends React.PureComponent<Props> {
+class AlertChart extends AsyncComponent<Props, State> {
+  componentDidUpdate(prevProps: Props) {
+    const {project, organization, start, end, period, utc} = this.props;
+
+    if (
+      prevProps.start !== start ||
+      prevProps.end !== end ||
+      prevProps.period !== period ||
+      prevProps.utc !== utc ||
+      prevProps.organization.id !== organization.id ||
+      prevProps.project.id !== project.id
+    ) {
+      this.remountComponent();
+    }
+  }
+
+  getDefaultState(): State {
+    return {
+      ...super.getDefaultState(),
+      alerts: [],
+    };
+  }
+
+  getEndpoints(): ReturnType<AsyncComponent['getEndpoints']> {
+    const {project, organization, period, start, end, utc, rule} = this.props;
+
+    return [
+      [
+        'alerts',
+        `/projects/${organization.slug}/${project.slug}/rules/${rule.id}/stats/`,
+        {
+          query: {
+            ...(period && {statsPeriod: period}),
+            start,
+            end,
+            utc,
+          },
+        },
+      ],
+    ];
+  }
   renderChart() {
-    const TOTAL = 6;
-    const NOW = new Date().getTime();
-    const getValue = () => Math.round(Math.random() * 1000);
-    const getDate = num => NOW - (TOTAL - num) * 86400000;
-    const getData = num =>
-      [...Array(num)].map((_v, i) => ({value: getValue(), name: getDate(i)}));
+    const {alerts} = this.state;
```

Comment:
alerts feels vague on this page since we're in alerts section and in an alert rule. The backend called it `RuleFireHistory`